buildkite/bazel-bench/generate_report.py

```diff
@@ -0,0 +1,259 @@
+#!/usr/bin/env python3
+#
+# Copyright 2019 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Generates a daily HTML report for the projects.
+The steps:
+  1. Get the necessary data from Storage for projects/date.
+  2. Manipulate the data to a format suitable for graphs.
+  3. Generate a HTML report containing the graphs.
+  4. Upload the generated HTMLs to GCP Storage.
+"""
+import argparse
+import collections
+import csv
+import datetime
+import json
+import io
+import os
+import statistics
+import subprocess
+import sys
+import tempfile
+import urllib.request
+
+
+def _platform_path_str(posix_path):
+    """Converts the path to the appropriate format for platform."""
+    if os.name == "nt":
+        return posix_path.replace("/", "\\")
+    return posix_path
+
+
+# For the event markers in JSON profiles.
+# TODO(leba): Include JSON profiles data.
+EVENTS_ORDER = [
```

Comment:
Why do you specify the order here explicitly? They should just be specified in this order in the file, right?

Reply:
Forgot to remove this. Removed as it is irrelevant in this PR.